src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -2,76 +2,164 @@
+import { ChromeRecentlyAccessedHistoryItem, SavedObject } from 'opensearch-dashboards/public';
 import { WorkspaceObject } from '../../../workspace';
-import { createRecentNavLink } from './nav_link';
 import { HttpStart } from '../../../http';
-import { ChromeNavLink } from '../../../';
 import './recent_items.scss';
 
+const widthForRightMargin = 8;
+
 export interface Props {
   recentlyAccessed$: Rx.Observable<ChromeRecentlyAccessedHistoryItem[]>;
   workspaceList$: Rx.Observable<WorkspaceObject[]>;
   navigateToUrl: (url: string) => Promise<void>;
-  basePath: HttpStart['basePath'];
-  navLinks$: Rx.Observable<ChromeNavLink[]>;
   renderBreadcrumbs: React.JSX.Element;
   buttonSize?: EuiHeaderSectionItemButtonProps['size'];
+  http: HttpStart;
+}
+
+interface SavedObjectMetadata {
+  icon?: string;
+  title?: string;
+  editUrl?: string;
+  inAppUrl?: { path: string; uiCapabilitiesPath: string };
+  namespaceType?: SavedObjectsNamespaceType;
 }
+export type SavedObjectWithMetadata<T = unknown> = SavedObject<T> & {
+  meta: SavedObjectMetadata;
+};
+
+type DetailedRecentlyAccessedItem = ChromeRecentlyAccessedHistoryItem &
+  SavedObjectWithMetadata &
+  ChromeRecentlyAccessedHistoryItem['meta'] & {
+    updatedAt: number;
+    workspaceName?: string;
+  };
+
+const bulkGetDetail = (savedObjects: Array<Pick<SavedObject, 'type' | 'id'>>, http: HttpStart) => {
+  return Promise.all(
+    savedObjects.map((obj) =>
+      http
+        .get<SavedObjectWithMetadata>(
+          `/api/opensearch-dashboards/management/saved_objects/${encodeURIComponent(
+            obj.type
+          )}/${encodeURIComponent(obj.id)}`
+        )
+        .catch((error) => ({
+          id: obj.id,
+          type: obj.type,
+          error,
+          attributes: {},
+          references: [],
+          meta: {},
+          updated_at: '',
+        }))
+    )
+  );
+};
+
+const recentsRadios = [
+  {
+    id: '5',
+    label: '5 pages',
+  },
+  {
+    id: '10',
+    label: '10 pages',
+  },
+  {
+    id: '15',
+    label: '15 pages',
+  },
+];
 
 export const RecentItems = ({
   recentlyAccessed$,
   workspaceList$,
   navigateToUrl,
-  navLinks$,
-  basePath,
   renderBreadcrumbs,
   buttonSize = 's',
+  http,
 }: Props) => {
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
-
+  const [isPreferencesPopoverOpen, setIsPreferencesPopoverOpen] = useState(false);
   const recentlyAccessedItems = useObservable(recentlyAccessed$, []);
+  const [recentsRadioIdSelected, setRecentsRadioIdSelected] = useState(recentsRadios[0].id);
   const workspaceList = useObservable(workspaceList$, []);
-  const navLinks = useObservable(navLinks$, []).filter((link) => !link.hidden);
-
-  const items = useMemo(() => {
-    // Only display five most recent items
-    return recentlyAccessedItems.slice(0, 5).map((item) => {
-      return {
-        link: createRecentNavLink(item, navLinks, basePath, navigateToUrl).href,
```

Comment:
Seems we removed the createRecentNavLink import, but the link should be generated from it.

Reply:
Hi, I delete it since we could directly access link from recentlyAccessedItems

Reply:
But that link is not correct as it does not contain workspace info.

Reply:
ok,I see, I will correct this. Thank you